### PR TITLE
feat(rpc): impl syncing call

### PR DESCRIPTION
**Pull request by mattsse**
impl Syncing handler.

unsure about the different blocks because this is a bit hard to differentiate with staged sync.

### CODE REVIEW

File: crates/net/rpc/src/eth/api/server.rs [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
 
     fn syncing(&self) -> Result<SyncStatus> {
-        Err(internal_rpc_err("unimplemented"))
+        if self.network().is_syncing() {
+            let info = self.client().chain_info().to_rpc_result()?;
+            let best = U256::from(info.best_number);
+            Ok(SyncStatus::Info(SyncInfo {
+                starting_block: U256::ZERO,
+                current_block: best,
+                highest_block: best,
```
Comment (rkrasiuk) [inline crates/net/rpc/src/eth/api/server.rs:44]: can't find any more clarification on `highest_block`. i assume [`The estimated highest block`](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_syncing) means the observed tip from CL?

Comment (onbjerg) [reply crates/net/rpc/src/eth/api/server.rs:44]: Yes, it is what we think is the highest block (our sync target)

File: crates/net/rpc/src/eth/api/server.rs [hunk 2]
```diff
@@ -1,40 +1,53 @@
...
     }
 
     fn syncing(&self) -> Result<SyncStatus> {
-        Err(internal_rpc_err("unimplemented"))
+        if self.network().is_syncing() {
+            let info = self.client().chain_info().to_rpc_result()?;
+            let best = U256::from(info.best_number);
+            Ok(SyncStatus::Info(SyncInfo {
+                starting_block: U256::ZERO,
+                current_block: best,
```
Comment (onbjerg) [inline crates/net/rpc/src/eth/api/server.rs:43]: This block should correspond to whatever is passed to https://github.com/paradigmxyz/reth/blob/b178631bc17708ebc85043dfef99adac473c6211/crates/net/network/src/network.rs#L229

File: crates/net/rpc/src/eth/api/server.rs [hunk 3]
```diff
@@ -1,40 +1,53 @@
...
         EthApiSpec::protocol_version(self).await.to_rpc_result()
     }
 
     fn syncing(&self) -> Result<SyncStatus> {
-        Err(internal_rpc_err("unimplemented"))
+        if self.network().is_syncing() {
+            let info = self.client().chain_info().to_rpc_result()?;
+            let best = U256::from(info.best_number);
+            Ok(SyncStatus::Info(SyncInfo {
+                starting_block: U256::ZERO,
```
Comment (onbjerg) [inline crates/net/rpc/src/eth/api/server.rs:42]: This should ideally correspond to the first block we process in the pipeline on start, so either genesis or whatever the `Finish` stage wrote to the database. I would mark this with a note

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Pull request closed without merging**